src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    const { keysToRemove, extraData, type } = action;
+
+    switch (true) {
+      case type === 'addProperties':
+        Object.assign(copyState, extraData);
```

Comment:
```suggestion
    switch (type) {
      case 'addProperties':
        Object.assign(copyState, extraData);
```
Inappropriate usage of switch

Reply:
[UL](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch)